pkg/client/client.go

```diff
@@ -298,13 +299,25 @@ func NewTokenAuthClient(token string) *http.Client {
 
 // NewOAuthClient creates a new HTTP client that supports OAuth2 client credentials based authorization
 func NewOAuthClient(oauthConfig OauthCredentials) *http.Client {
+
+	config := getClientCredentialsConfig(oauthConfig)
+	return config.Client(context.TODO())
+}
+
+func getClientCredentialsConfig(oauthConfig OauthCredentials) clientcredentials.Config {
```

Comment:
discussion/todo: try reverting this and testing on an actual HTTP client integration level, this is safer to changes